libs/build_met.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash                                                                                                                              
+
+set -eux
+
+name="met"
+version=${1:-${STACK_met_version}}
+release_date=${2:-${STACK_met_release_date}}
+
+cd  ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}
+software=$name-$version.$release_date
+pkg_name=$name-$version
+url="https://github.com/dtcenter/MET/releases/download/v$version/$software.tar.gz"
+[[ -d $software ]] || ( $WGET $url; tar -xf $software.tar.gz )
+[[ -d $pkg_name ]] && cd $pkg_name || ( echo "$pkg_name does not exist, ABORT!"; exit 1 )
+
+
+# Hyphenated version used for install prefix
+compiler=$(echo $HPC_COMPILER | sed 's/\//-/g')
+mpi=$(echo $HPC_MPI | sed 's/\//-/g')
+
+export LMOD_EXACT_MATCH="no"
+
+if $MODULES; then
+    
+    set +x
+    source $MODULESHOME/init/bash
+    module load hpc-$HPC_COMPILER
+    [[ -z $mpi ]] || module load hpc-$HPC_MPI
+    module load hdf5
+    module load netcdf
+    module load bufr
+    module load g2c
+    module load zlib
+    module load jasper
+    module load png
+    module load gsl
+    module list
+    set -x
+    prefix="${PREFIX:-"/opt/modules"}/$compiler/$mpi/$name/$version"
+    if [[ -d $prefix ]]; then
+	[[ $OVERWRITE =~ [yYtT] ]] && ( echo "WARNING: $prefix EXISTS: OVERWRITING!";$SUDO rm -rf $prefix ) \
+	    || ( echo "WARNING: $prefix EXISTS, SKIPPING"; exit 1 )
+    fi
+else
+
+    prefix=${MET_ROOT:-"/usr/local"}
+
+fi
+
+export MET_BASE=$prefix/share/met
+
+if [[ ! -z $mpi ]]; then
+    export FC=$MPI_FC
+    export CC=$MPI_CC
+    export CXX=$MPI_CXX
+else
+    export FC=$SERIAL_FC
+    export CC=$SERIAL_CC
+    export CXX=$SERIAL_CXX
+fi
+
+export F77=$FC
+export FFLAGS="${STACK_FFLAGS:-} ${STACK_met_FFLAGS:-}"
+export CFLAGS="${STACK_CFLAGS:-} ${STACK_met_CFLAGS:-}"
+
+export MET_NETCDF=${NetCDF_ROOT}
+export MET_HDF5=${HDF5_ROOT}
+export MET_BUFRLIB=${bufr_ROOT}/lib
+export MET_GRIB2CLIB=${g2c_ROOT}/lib
+export MET_GRIB2CINC=${g2c_ROOT}/include
+export MET_GSL=${GSL_ROOT}
+export BUFRLIB_NAME=-lbufr_4
+export GRIB2CLIB_NAME=-lg2c
+export LIB_JASPER=${JASPER_ROOT}/lib64
+export LIB_LIBPNG=${PNG_ROOT}/lib64
+export LIB_Z=${ZLIB_ROOT}/lib
+
+LDFLAGS1="-Wl,--disable-new-dtags"
+LDFLAGS2="-Wl,-rpath,${MET_NETCDF}/lib:${MET_HDF5}/lib:${MET_BUFRLIB}"
+LDFLAGS3="-Wl,-rpath,${MET_GRIB2CLIB}:${MET_PYTHON}/lib:${MET_GSL}/lib"
+LDFLAGS4="-L${LIB_JASPER} -L${MET_HDF5}/lib -L${LIB_LIBPNG} -L${LIB_Z}"
+LDFLAGS5="-L${I_MPI_ROOT}/lib64"
```

Comment:
I doubt this will work on a generic machine.

Reply:
@aerorahul 
Can you please explain why?  This is in my install script where I have installed MET on WCOSS, Hera, Stampede, Orion, Jet, Cheyenne, and local machines and it works on all of them.  Although 
> LDFLAGS5="-L${I_MPI_ROOT}/lib64"
is not in my script that I use and I don't recall adding it here, although I'm guessing it was necessary to get it to build on WCOSS.